src/Commands/DrupalCommands.php

```diff
@@ -47,6 +48,26 @@ public function setRuntimeConfig(ConsoleCommandEvent $event)
         }
     }
 
+    /**
+     * @hook validate drupal:site-install
+     *
+     * @param CommandData $commandData
+     * @throws \Exception
+     */
+    public function validateSiteInstall(CommandData $commandData)
+    {
+        $input = $commandData->input();
+        $siteDirectory = getcwd().'/'.$input->getOption('root').'/sites/'.$input->getOption('sites-subdir');
+
+        // Check that if files/folders required exist and they are writable.
+        $requiredFiles = [$siteDirectory, $siteDirectory.'/settings.php'];
+        foreach ($requiredFiles as $requiredFile) {
+            if (file_exists($requiredFile) && !is_writable($requiredFile)) {
+                throw new \Exception("The file/folder '$requiredFile' must be writable for installation to continue.");
```

Comment:
sprintf('The file/folder '%s' must be writable for installation to continue.', $requiredFile);